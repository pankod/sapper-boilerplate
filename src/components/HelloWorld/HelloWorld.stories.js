import HelloWorld from "./index";

export default {title: "HelloWorld"}

export const basic = () => ({
    Component: HelloWorld
});
