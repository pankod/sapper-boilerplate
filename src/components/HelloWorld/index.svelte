<style src="./style.scss">

</style>

<div class="title">
    <span class="title__top">Hello</span>
    <span class="title__front">World</span>
    <span class="title__back">World</span>
</div>
