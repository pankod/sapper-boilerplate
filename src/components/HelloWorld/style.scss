.title {
  font-family: Arial, Helvetica, sans-serif;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  position: relative;
  z-index: 1;
  text-align: center;

  &__top {
      font-size: 110px;
      color: #ffffff;
      margin: 0 0 10px;
  }

  &__back {
    color: rgba(255, 255, 255, 0.15);
    font-size: 140px;
    font-weight: 800;
    line-height: 86px;
  }

  &__front {
    color: #ffffff;
    font-size: 80px;
    font-weight: 500;
    position: absolute;
    margin-top: 75px;
  }
}
