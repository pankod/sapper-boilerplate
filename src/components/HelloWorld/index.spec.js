import HelloWorld from './index.svelte';
import { render } from '@testing-library/svelte'

describe('HelloWorld', () => {
    it("matches snapshot", () => {
        const { container } = render(HelloWorld);
        expect(container).toMatchSnapshot();
    });
    
    it('renders Hello World! text', () => {
        const { container, getAllByText } = render(HelloWorld);

        expect(getAllByText('Hello').length).toBe(1);
        expect(getAllByText('World').length).toBe(2);

        expect(container.getElementsByClassName('title').length).toBe(1);
        expect(container.getElementsByClassName('title__top').length).toBe(1);
        expect(container.getElementsByClassName('title__front').length).toBe(1);
        expect(container.getElementsByClassName('title__back').length).toBe(1);
    });

});
