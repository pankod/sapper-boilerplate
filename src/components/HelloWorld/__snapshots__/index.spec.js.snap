// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World renders Hello World! text 1`] = `
<body>
  <div>
    <div
      class="title"
    >
      <span
        class="title__top"
      >
        Hello
      </span>
       
      <span
        class="title__front"
      >
        World
      </span>
       
      <span
        class="title__back"
      >
        World
      </span>
    </div>
  </div>
</body>
`;
